llvm/include/llvm/ADT/Triple.h

```diff
@@ -211,7 +214,7 @@ class Triple {
     CoreCLR,
     Simulator,  // Simulator variants of other systems, e.g., Apple's iOS
     SYCLDevice,
-    LastEnvironmentType = Simulator
+    LastEnvironmentType = SYCLDevice
```

Comment:
Was it a bug or forgotten?

Reply:
I think when they added the SYCLDevice triple they forgot to update the LastEnvironmentType, so it appears to be a bug/something that was forgotten when added. Last_Type should always point to the last element.